### PR TITLE
fix: Fix inheritance session issues

### PR DESCRIPTION
**Pull request by TejasvOnly**

### CODE REVIEW

File: common/libraries/crypto/bip32.c [hunk 1]
```diff
@@ -757,75 +809,65 @@ int hdnode_nem_decrypt(const HDNode *node, const ed25519_public_key public_key,
...
+        node->curve->params, node->private_key, digest, sig, pby, is_canonical);
   } else if (node->curve == &curve25519_info) {
-    return 1;  // signatures are not supported
+    return 1;    // signatures are not supported
   } else {
     return hdnode_sign(node, digest, 32, 0, sig, pby, is_canonical);
   }
 }
 
-int hdnode_get_shared_key(const HDNode *node, const uint8_t *peer_public_key,
```
Comment (vrockz747) [inline common/libraries/crypto/bip32.c:802]: why is this function removed?

File: common/libraries/crypto/bip32.c [hunk 2]
```diff
@@ -757,75 +809,65 @@ int hdnode_nem_decrypt(const HDNode *node, const ed25519_public_key public_key,
...
-    }
-    curve25519_scalarmult(session_key + 1, node->private_key,
-                          peer_public_key + 1);
-    *result_size = 33;
-    return 0;
-  } else {
-    *result_size = 0;
-    return 1;  // ECDH is not supported
-  }
-}
```
Comment (vrockz747) [inline common/libraries/crypto/bip32.c:825]: is this delete intentional? why is this deleted?

Comment (TejasvOnly) [reply common/libraries/crypto/bip32.c:825]: Not used anywhere, seems like an old function which was never used.

**Review by vrockz747 — APPROVED** [review]
